app/utils/ContributionIdGenerator.scala

```diff
@@ -0,0 +1,10 @@
+package utils
+import java.util.UUID
+
+trait ContributionIdGenerator {
+  def getNewId: String
+}
+
+object ContributionIdGeneratorImpl extends ContributionIdGenerator {
```

Comment:
It looks like the trait isn't used here.  Unless you are using this for testing (which I don't think you are) it might be worth simply calling the object rather than injecting it into the controller then removing the trait.